lib/ui/window/key_data_packet.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_WINDOW_KEY_DATA_MESSAGE_H_
+#define FLUTTER_LIB_UI_WINDOW_KEY_DATA_MESSAGE_H_
+
+#include <string.h>
+#include <vector>
+#include <functional>
+
+#include "flutter/fml/macros.h"
+#include "flutter/lib/ui/window/key_data.h"
+
+namespace flutter {
+
+typedef std::function<void (bool)> KeyDataPacketCallback;
+
+class KeyDataPacket {
+ public:
+  // Build a KeyDataPacket by incrementally fill in data.
+  //
+  // The `character_data_size` is number of bytes to contain the character data.
+  KeyDataPacket(size_t character_data_size);
+  ~KeyDataPacket();
+
+  const std::vector<uint8_t>& data() const { return data_; }
+
+  void SetKeyData(const KeyData& event);
+
+  // Set character data to the proper position, which should not be terminated
+  // by a null character (length controled by character_data_size).
```

Comment:
Having setting the length and setting the data being completely different operations seems dangerous. What's the purpose of this? The only use I see in the PR of constructing one of these objects all the data is set at the same time, so it seems like it would be much, much simpler from an API perspective to just take everything in the constructor. Is there another use case for this API?